### ISSUE TITLE
Terraform apply is stuck with lost node

### ISSUE THREAD

**Issue by vaghetti** [rancher/terraform-provider-rke#211]
I created a 5 node cluster using this module, then I lost one of the nodes. I tried to replace the node that was lost by just swapping the IPs in the terraform config file, but it seems that terraform apply just sits there waiting forever. 

I do see an etcd container running in the new node, so the provider did manage to connect to the node, but it is still not joining the cluster.

Any suggestions on what I could do here? Is this expected behavior?

**Comment by rawmind0** [comment]
Could you please provide your tf file??

**Comment by vaghetti** [comment]
Hello @rawmind0 

This is the original resource: 
```resource rke_cluster "cluster" {
  nodes {
    address          = "34.68.159.119"
    internal_address = "10.128.0.22"
    user             = "rke"
    role             = ["controlplane", "etcd", "worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "35.193.142.175"
    internal_address = "10.128.0.23"
    user             = "rke"
    role             = ["controlplane", "etcd", "worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "35.223.177.202"
    internal_address = "10.128.0.24"
    user             = "rke"
    role             = ["controlplane", "etcd", "worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "34.69.101.223"
    internal_address = "10.128.0.26"
    user             = "rke"
    role             = ["worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "35.224.66.130"
    internal_address = "10.128.0.25"
    user             = "rke"
    role             = ["worker"]
    ssh_key          = var.rke_ssh_private_key
  }
}
```


This is the new version with the IP of the new node and the old one removed:

```
resource rke_cluster "cluster" {
  nodes {
    address          = "34.68.159.119"
    internal_address = "10.128.0.22"
    user             = "rke"
    role             = ["controlplane", "etcd", "worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "35.193.142.175"
    internal_address = "10.128.0.23"
    user             = "rke"
    role             = ["controlplane", "etcd", "worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "34.69.101.223"
    internal_address = "10.128.0.26"
    user             = "rke"
    role             = ["worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "35.224.66.130"
    internal_address = "10.128.0.25"
    user             = "rke"
    role             = ["worker"]
    ssh_key          = var.rke_ssh_private_key
  }
  nodes {
    address          = "104.198.41.161"
    internal_address = "10.128.0.27"
    user             = "rke"
    role             = ["controlplane", "etcd", "worker"]
    ssh_key          = var.rke_ssh_private_key
  }
}
```

**Comment by rawmind0** [comment]
Hi @vaghetti, could you please configure the provider to save rke logs, apply again and provide rke logs??
```
provider "rke" {
  debug = true
  log_file = "rke_debug.log"
}
```

**Comment by vaghetti** [comment]
Hello @rawmind0. I ended up deleting the cluster and creating a new one, but I did get it working again so I'll share what I did in case someone has the same issue:

It seemed like the problem was related to the node IPs getting mixed up with the `hostname_override`. I initially created them with a set of nodes without specifying hostname_override for them. When I lost some nodes, I simply replaced the IPs in the nodes I had on terraform. The nodes that got their IP swapped preserved the old IPs as `hostname_override`, but got new IPs assigned and that apparently confused the rke installer.

Another thing that I had to do to get it working again was `kubectl delete node` on the master node that was replaced.

Closing the issue.
